src/Xamarin.Android.Build.Tasks/Tasks/AndroidDotnetToolTask.cs

```diff
@@ -27,12 +36,22 @@ public override bool Execute ()
 
 			var assemblyPath = Path.Combine (ToolPath, $"{BaseToolName}.dll");
 			if (File.Exists (assemblyPath)) {
-				Log.LogDebugMessage ($"Using: dotnet {assemblyPath}");
 				UsingDotnet = true;
 				AssemblyPath = assemblyPath;
 				ToolPath = null;
+
+				Log.LogDebugMessage ($"Using: dotnet {AssemblyPath}");
 			} else {
-				Log.LogDebugMessage ($"Using: {GenerateFullPathToTool ()}");
+				if (!RuntimeInformation.IsOSPlatform (OSPlatform.Windows) &&
+						!RuntimeInformation.FrameworkDescription.StartsWith ("Mono", StringComparison.OrdinalIgnoreCase)) {
+					UsingMono = true;
+					AssemblyPath = Path.ChangeExtension (assemblyPath, ".exe");
+					ToolPath = null;
+
+					Log.LogDebugMessage ($"Using: mono {AssemblyPath}");
+				} else {
+					Log.LogDebugMessage ($"Using: {GenerateFullPathToTool ()}");
```

Comment:
Checking a build log on mac, it seems like we are still hitting the preferred path running on Mono:
```
Using: /Users/builder/azdo/_work/1/s/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/generator.exe
```